config.template

```diff
@@ -1,88 +1,117 @@
 seaice_streams_filename = streams.cice
-
-[paths]
-# paths to simulation and observational datasets
-archive_dir = /scratch1/scratchdirs/petercal/ACME_simulations
-archive_dir_ocn = /scratch1/scratchdirs/petercal/ACME_simulations/20160805v0atm.A_WCYCL1850_v0atm.ne30_oEC.edison.alpha7_00/run
-scratch_dir = /global/project/projectdirs/acme/xylar/20160805v0atm.A_WCYCL1850_v0atm.ne30_oEC.edison.alpha7_00.test.pp
-plots_dir = /global/project/projectdirs/acme/xylar/coupled_diagnostics_20160805v0atm.A_WCYCL1850_v0atm.ne30_oEC.edison.alpha7_00-20160520.A_WCYCL1850.ne30_oEC.edison.alpha6_01
-log_dir = /global/project/projectdirs/acme/xylar/coupled_diagnostics_20160805v0atm.A_WCYCL1850_v0atm.ne30_oEC.edison.alpha7_00-20160520.A_WCYCL1850.ne30_oEC.edison.alpha6_01.logs
-obs_ocndir = /global/project/projectdirs/acme/observations/Ocean
-obs_sstdir = /global/project/projectdirs/acme/observations/Ocean/SST
-obs_sssdir = /global/project/projectdirs/acme/observations/Ocean/SSS
-obs_mlddir = /global/project/projectdirs/acme/observations/Ocean/MLD
-obs_seaicedir = /global/project/projectdirs/acme/observations/SeaIce
-ref_archive_v0_ocndir = /global/project/projectdirs/acme/ACMEv0_lowres/B1850C5_ne30_v0.4/ocn/postprocessing
-ref_archive_v0_seaicedir = /global/project/projectdirs/acme/ACMEv0_lowres/B1850C5_ne30_v0.4/ice/postprocessing
-
-[data]
-# paths to mesh and mapping files
+# analysis_ref_date is the starting date of the simulation according to the
+#    analysis.  The starting date of the simulation acccording to the input
+#    files will be read in from the variable simulationStartTime.  If
+#    simulationStartTime is different from analysis_ref_date, an offset will
+#    be added to all dates when they are read in.  This would be desirable,
+#    for example, if the simulation year 0001 should correspond with analysis
+#    year 1850.
+#
+# If no analysis offset is desired, simply comment out or remove
+# analysis_ref_date from this section.
+analysis_ref_date = 1850-01-01
+
+[output]
+
+# directory where analysis should be written
+basedir = /dir/to/analysis/output
+
+# subdirectories within basedir for analysis output
+scratch_subdir = scratch
+plots_subdir = plots
+logs_subdir = logs
+clim_subdir = clim
+
+# a list of which analyses to generate.  Valid names are:
+#   'ohc_timeseries', 'sst_timeseries', 'sst_modelvsobs', 'sss_modelvsobs',
+#   'mld_modelvsobs', 'seaice_timeseries', 'seaice_modelvsobs'
+# the following shortcuts exist:
+#   'all' -- all analyses will be run
+#   'all_timeseries' -- all time-series analyses will be run
+#   'all_modelvsobs' -- all analyses comparing model climatologies with
+#                       observatoins will be run
+#   'all_ocean' -- all ocean analyses will be run
+#   'all_seaice' -- all sea-ice analyses will be run
+generate = ['all']
+
+# an alternative example that would perform all analysis except
+#   'ohc_timeseries'
+#generate = ['sst_timeseries', 'all_modelvsobs', 'all_seaice']
```

Comment:
@pwolfram, I think your idea is a good one.  I wasn't thinking of my example as being a good way of doing all analysis other than `ohc_timeseries`, just that it happened to work out that way.  But I can see how it came across that way.

Anyway, I agree that there needs to be a way of explicitly turning off an analysis function.  The only problem I have with the `-ohc_timeseries` syntax is that I was thinking of having a similar syntax to `generate` on the command line, for example:
```
./run_analysis config.analysis --generate sst_timeseries,all_modelvsobs,all_seice
```
(I'm open to other command line syntax).  I think it might be a bit confusing to have the `-ohc_timeseries` syntax on the command line (or it might fail completely).  Would you be open to `no_ohc_timeseries` (as well as `no_ocean`, `no_timeseries`, etc.) as alternatives to the minus sign?  This would be consistent with the approach used in `run_test.py --no_init_step1 --no_forward` syntax used in the MPAS testcases infrastructure (soon to be renamed COMPASS).